src/main/java/lol/server/Server.java

```diff
@@ -41,81 +34,104 @@ public void run() {
     System.out.println("Server shutted down.");
   }
 
+  public void registrationPhase() throws IOException {
+    Scanner scanner = new Scanner(System.in);
+    System.out.println("Enter the number of participants for the tourney/match: ");
+    int expectedParticipants = scanner.nextInt();
```

Comment:
Optional modification: It would be better if the number of players were passed an argument to the program, e.g., retrieved in `args` of the main function. This will simplify the automation of the tournament.